publisher/internal/analytics/ethereum/convert.go

```diff
@@ -110,3 +110,22 @@ func splitMintDatatoHexFields(data string) (string, string, string, error) {
 
 	return amountHex, amountToken0Hex, amountToken1Hex, nil
 }
+
+func splitCollectDatatoHexFields(data string) (string, string, string, error) {
```

Comment:
Can the argument parsing be done using ABI with the help of `go-ethereum`?

Reply:
Tried to introduce ABIs. Probably might need some improvement with file placements or namings.